contracts/prize-pool/PrizePool.sol

```diff
@@ -108,72 +186,79 @@ abstract contract PrizePool is Initializable, AbstractYieldService, BaseRelayRec
     _redeem(amountLessFee);
     _token().transfer(from, amountLessFee);
 
-    comptroller.afterWithdrawInstantlyFrom(operator, from, amount, token, exitFee, sponsoredExitFee);
+    if (hasPrizeStrategy) {
+      prizeStrategy.afterWithdrawInstantlyFrom(operator, from, amount, token, exitFee, sponsoredExitFee);
+    }
 
     emit InstantWithdrawal(operator, from, token, amount, exitFee, sponsoredExitFee);
-
-    // return the exit fee
-    return exitFee;
   }
 
+  /// @notice Withdraw assets from the Prize Pool with a timelock on the assets
+  /// @dev The timelock is used to ensure that the tickets have contributed their equal weight
+  /// in the Prize before being withdrawn, in order to prevent gaming the system
+  /// @param from The address to withdraw assets from by redeeming tickets
+  /// @param amount The amount of assets to redeem for tickets
+  /// @param token The address of the asset token being withdrawn
+  /// @return unlockTimestamp The unlock timestamp that the assets will be released upon
   function withdrawWithTimelockFrom(
     address from,
     uint256 amount,
     address token
   )
-    external nonReentrant onlyControlledToken(token) returns (uint256)
+    external
+    nonReentrant
+    onlyControlledToken(token)
+    returns (uint256 unlockTimestamp)
   {
     _updateAwardBalance();
 
-
     address operator = _msgSender();
-
     ControlledToken(token).controllerBurnFrom(operator, from, amount);
 
     // Sweep the old balance, if any
     address[] memory users = new address[](1);
     users[0] = from;
-
-
     sweepTimelockBalances(users);
 
-
     timelockTotalSupply = timelockTotalSupply.add(amount);
     timelockBalances[from] = timelockBalances[from].add(amount);
 
-    uint256 unlockTimestamp = comptroller.calculateWithdrawalUnlockTimestamp(from, amount, token);
+    // the block at which the funds will be available
+    if (_hasPrizeStrategy()) {
+      unlockTimestamp = prizeStrategy.calculateWithdrawalUnlockTimestamp(from, amount, token);
```

Comment:
I'm sure you'll see that `calculateWithdrawalUnlockTimestamp` has now changed to `beforeWithdrawWithTimelock`

It needs to run *before* any state changes in the PrizePool, so you'll have to separate these like you did above